### PR TITLE
Move VD scanner and database feed manager inner the engine project compilation

### PR DESCRIPTION
**Pull request by Dwordcito**
|Related issue|
|---|
|#25106|

This pr aims to:
- Move vulnerability scanner and feed manager to compile the project inner the engine workflow.
- Unit tests of the scanner.
- Refactor efficacy test execution.
- Remove unused workflows.

### CODE REVIEW

**Review by sebasfalcone — COMMENTED** [review]
Question: Shouldn't the directory `wazuh/src/wazuh_modules` be removed?

File: src/engine/CMakeLists.txt [hunk 1]
```diff
@@ -160,13 +160,17 @@ find_and_create_imported_target("opentelemetry-cpp" "opentelemetry-cpp::logs")
...
+find_and_create_imported_target("httplib" "httplib::httplib")
 
 ####################################################################################################
 # Targets
 ####################################################################################################
 # Build main
 add_executable(main ${ENGINE_SOURCE_DIR}/main.cpp)
 
+add_subdirectory(${ENGINE_SOURCE_DIR}/feedmanager)
+add_subdirectory(${ENGINE_SOURCE_DIR}/vdscanner)
```
[the remark this answers is not in the document]
Comment (Dwordcito) [reply src/engine/CMakeLists.txt:173]: yes

File: src/engine/source/vdscanner/qa/test_false_negative_data/001/input_001.json
[the remark this answers is not in the document]
Comment (Dwordcito) [reply src/engine/source/vdscanner/qa/test_false_negative_data/001/input_001.json:1]: Current clangformat remove the new line, we can iterate over this later.

File: .github/workflows/engine_vdscanner_efficacy.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    timeout-minutes: 60
+
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v3
+
+    - name: Cache CMake and ccache
+      uses: hendrikmuhs/ccache-action@v1.2
+      with:
+        key: ${{ github.workflow }}-${{ runner.os }}
```
[the remark this answers is not in the document]
Comment (Dwordcito) [reply .github/workflows/engine_vdscanner_efficacy.yml:50]: yeap, we will create a new issue to fix this.

File: src/engine/source/vdscanner/qa/test_false_positive_data/002/input_001.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    "major_version": "10",
+    "minor_version": "0",
+    "build": "19045.4043",
+    "name": "Microsoft Windows 10 Pro",
+    "display_version": "22H2",
+    "platform": "windows",
+    "version": "10.0.19045.4043",
+    "scan_time": "2023/08/04 19:56:11"
+  }
+}
```
[the remark this answers is not in the document]
Comment (Dwordcito) [reply src/engine/source/vdscanner/qa/test_false_positive_data/002/input_001.json:40]: dito

File: src/engine/source/vdscanner/qa/test_false_positive_data/003/input_001.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    "major_version": "10",
+    "minor_version": "0",
+    "build": "19045.4291",
+    "name": "Microsoft Windows 10 Pro",
+    "display_version": "22H2",
+    "platform": "windows",
+    "version": "10.0.19045.4291",
+    "scan_time": "2024/04/24 14:00:00"
+  }
+}
```
[the remark this answers is not in the document]
Comment (Dwordcito) [reply src/engine/source/vdscanner/qa/test_false_positive_data/003/input_001.json:39]: dito

**Review by sebasfalcone — APPROVED** [review]
All inquires answered, LGTM

**Pull request merged**